### PR TITLE
Add MetricInfo class and tests

### PR DESCRIPTION
**Pull request by mziccard**

### CODE REVIEW

File: gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    assertEquals(DESCRIPTION, METRIC_INFO.description());
+  }
+
+  @Test
+  public void testToBuilder() {
+    compareMetricInfo(METRIC_INFO, METRIC_INFO.toBuilder().build());
+    MetricInfo metricInfo = METRIC_INFO.toBuilder()
+        .name("newName")
+        .description("newDescription")
+        .filter("logName=projects/my-projectid/logs/syslog")
```
Comment (lesv) [inline gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java]: Shouldn't this be FILTER?

File: gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        .build();
+    assertEquals("newName", metricInfo.name());
+    assertEquals("logName=projects/my-projectid/logs/syslog", metricInfo.filter());
```
Comment (lesv) [inline gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java]: Shouldn't this also be FILTER?

File: gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    assertEquals(FILTER, METRIC_INFO.filter());
+    assertEquals(DESCRIPTION, METRIC_INFO.description());
+  }
+
+  @Test
+  public void testToBuilder() {
+    compareMetricInfo(METRIC_INFO, METRIC_INFO.toBuilder().build());
+    MetricInfo metricInfo = METRIC_INFO.toBuilder()
+        .name("newName")
+        .description("newDescription")
```
Comment (lesv) [inline gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java]: Should this have a constant string that is reflected in #L58 as well?

File: gcloud-java-logging/src/main/java/com/google/cloud/logging/MetricInfo.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+    /**
+     * Sets an optional description for this metric. Used for documentation purpose.
+     */
+    public abstract Builder description(String description);
+
+    /**
+     * Sets an advanced logs filter. The value of the metric is the number of log entries that match
+     * this filter. Example: {@code logName=projects/my-projectid/logs/syslog AND severity>=ERROR}.
+     *
```
Comment (lesv) [inline gcloud-java-logging/src/main/java/com/google/cloud/logging/MetricInfo.java]: Should there be a shortcut method for specifying just a log("blah") that creates the logname=projects/my-projectid/logs/blah?

**Comment by lesv** [comment]
Consider my comments, update and/or commit.

File: gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-logging/src/test/java/com/google/cloud/logging/MetricInfoTest.java]: Here I am trying to test that the `toBuilder()` method and setter can updated update previous values. So I am using a different filter (and also name and description).

I will add a constants for these new values as well.

File: gcloud-java-logging/src/main/java/com/google/cloud/logging/MetricInfo.java (hunk 4 above)
Comment (mziccard) [inline gcloud-java-logging/src/main/java/com/google/cloud/logging/MetricInfo.java]: Opened #1061 for this

**Pull request merged**